doc/design

```diff
@@ -21,3 +21,11 @@ outside of `charmander` through a `daemon`.
 4. Refactoring some of this daemon logic into factory functions which can do
 this setup for us automatically to improve performance and make the code easier
 to read through.
+
```

Comment:
This doc may be superseded by something external to this repo. Haven't decided yet.